### ISSUE TITLE
Add SMSSecure

### ISSUE THREAD

**Issue by sedrubal** [nylira/prism-break#1267]
Are there any arguments, not to recommend [`SMSSecure`](https://github.com/SMSSecure/SMSSecure)? Otherwise I'll create a pull request.

`SMSSecure` is a forge of [`TextSecure`](https://github.com/WhisperSystems/TextSecure), without `Google API` as dependency (this is why we not recommend `TextSecure`; see #1140 and #552). But without `Google API` there is not problem, isn't it?

**Comment by Dostoi** [comment]
And you can find it on F-Droid : 
https://f-droid.org/repository/browse/?fdfilter=smssecure&fdid=org.smssecure.smssecure

They're planning on making F-Droid their main way of releasing the app

**Comment by vyp** [comment]
I can give a +1 here. Been watching the development of this (was called SecuredText before), and it also recently hit f-droid. The only possible problem I think is that it's a young (as in recent) fork, but go ahead with pull request if you want of course. I don't think it would be a waste time.

**Comment by strugee** [comment]
One issue I see with SMSSecure is that the SMS transport requires a key exchange. This is prone to user error and is also ugly from a UX perspective. Hence why TextSecure got rid of it.

**Comment by sedrubal** [comment]
I see...

And most likely they will have the same problems like TextSecure, that you can't export and revoke your keys and that you can't give your public key through another communication channel and that there will be no difference between writing with a person you validated and writing with a person of whom you have a invalid public key...

But I think it is better to use this app than writing plain SMS...

**Comment by vyp** [comment]
True but as sedrubal says I think that is supposed to be more of an intentional "feature", than a bug.

**Comment by strugee** [comment]
> True but as sedrubal says I think that is supposed to be more of an intentional "feature", than a bug.

(I assume you're replying to me?)

The fact that key exchange is a feature instead of a bug doesn't mean that we can ignore the fact that it's unideal when considering it for inclusion on PRISM Break.

**Comment by vyp** [comment]
> (I assume you're replying to me?)

Yes sorry. But how else can it be done? Is it possible? Because they can't remove support for SMS like textsecure because that's the whole point of the project.

I get your point though, in that it isn't good enough prism break if it has that. But I think it's useful enough to include. While you're right that prism break is all about user friendliness, there's still a wide variety of applications that prism break suggests. And I wouldn't say key exchange is at the bottom of the list in terms of ease of use.

Although maybe I could be wrong/biased. Thoughts?

**Comment by sedrubal** [comment]
:+1:

I think SMS isn't dead yet and sending an encrypted SMS is better than sending a plain SMS. But prism break doesn't recommend any SMS app yet, even if you have to send a SMS sometimes...

**Comment by ximex** [comment]
:+1:

**Comment by strugee** [comment]
> But how else can it be done? Is it possible? Because they can't remove support for SMS like textsecure because that's the whole point of the project.

To be perfectly frank, I don't see this project ever being included on PRISM Break, because I think it's fundamentally flawed. I think the OpenWhisperSystems guys got rid of TextSecure's SMS transport for a reason, and I think that trying to preserve it in a fork is doomed to fail. My impression is that you seem to be approaching this issue (consciously or otherwise) in a way that assumes SMSSecure has a right to be on PRISM Break. I'm approaching it from the opposite: I think SMSSecure should have to prove its worth. I feel like you're simply explaining _why_ SMSSecure has problems, instead of actually offering technical arguments explaining why I'm wrong. (Of course, this are just my personal feeling on the matter. Such are the limits of textual communication.)

IMO, Key exchange should be an implementation detail. The protocol should handle it transparently, not the user. This is because a) the user won't be able to fully understand key exchanges and b) will inevitably just trust whatever key they get, or some other equally bad scenario.

> I think SMS isn't dead yet and sending an encrypted SMS is better than sending a plain SMS.

Both parties still have to be on the same platform. At which point you might as well use TextSecure. I understand that's a very difficult position, because we don't currently list TextSecure, but I feel very strongly that SMSSecure is not the answer, and I would be uncomfortable recommending it.

> I get your point though, in that it isn't good enough prism break if it has that. But I think it's useful enough to include.

Friendly reminder that PRISM Break emphasizes quality over quantity.

> But how else can it be done? Is it possible? Because they can't remove support for SMS like textsecure because that's the whole point of the project.

**Comment by sedrubal** [comment]
>Friendly reminder that PRISM Break emphasizes quality over quantity.

OK, but thre is no SMS app on prism break. (It's like recommending to send plain SMS ;) )

> I think that trying to preserve it in a fork is doomed to
>fail.

Maybe you're right... We will see. For now It's working fine.

> we don't currently list TextSecure, but I feel very
>strongly that SMSSecure is not the answer

We primary don't list TextSecure because of Google dependency. SMSSecure is not an answer, it's a completely other thing.

**Comment by strugee** [comment]
> > Friendly reminder that PRISM Break emphasizes quality over quantity.

> OK, but thre is no SMS app on prism break. (It's like recommending to send plain SMS ;) )

See my remarks below about protocol security vs. medium security.

> > I think that trying to preserve it in a fork is doomed to fail.

> Maybe you're right... We will see. For now It's working fine.

I wasn't saying that it will fail on a technical level. I was saying that it will fail on a user experience level, and more importantly, I was saying that it _already has_ failed on a user experience level.

> > we don't currently list TextSecure, but I feel very strongly that SMSSecure is not the answer

> We primary don't list TextSecure because of Google dependency. SMSSecure is not an answer, it's a completely other thing.

But it is an answer. You're responding to the fact that we need something for the gap which TextSecure is supposed to fill. Yes, SMSSecure runs over a different protocol than TextSecure, but we shouldn't be securing protocols. We should be securing mediums. And SMSSecure and TextSecure are the same medium - instant messaging.

**Comment by sedrubal** [comment]
> _already has_ failed on a user experience level.

Why? Because it's a fork?

>we shouldn't be securing protocols.

Why?

>And SMSSecure and TextSecure are the same medium - instant messaging.

If you ask people, I don't think they will say, that SMS is instant messaging.

**Comment by sedrubal** [comment]
Are there still any arguments against SMSSecure?

BTW: Recently as TextSecure invented their new design, SMSSecure did this on the same day, so I think you can't say, they "failed on a user experience level"...

**Comment by vyp** [comment]
> Yes, SMSSecure runs over a different protocol than TextSecure, but we shouldn't be securing protocols. We should be securing mediums. And SMSSecure and TextSecure are the same medium - instant messaging.

I agree but smssecure does not require internet, which is probably why it's a special case/useful to list.

**Comment by juanito003** [comment]
I undersand strugee main argument not to list SMSSecure
>One issue I see with SMSSecure is that the SMS transport requires a key exchange. This is prone to user error and is also ugly from a UX perspective. Hence why TextSecure got rid of it.

Maybe there are more tech and IT people motivated enough to read prism-break website than general public. And those IT guys can advertise SMSSecure to there friends and family (at least I do). I guess it happens this way around, not the other one. And by doing this we can help understand the way key exchange work so friends and family can have little understanding about what to do to secure their SMS. And for a reason like this SMSSecure should be on the list.
What bad can happen if SMSSecure is not on the list? People don't know about it, but they know about TextSecure. Look at Schweineschwarte who commented 6 days ago on #1314. He was not aware SMSSecure existed and he is interested because he is still using TextSecure. And why was he not aware? Because SMSSecure was not on the prism-break list!

Many people have their android phone with GSM **ON** all day long. Not so true about wifi.

**Comment by strugee** [comment]
Hey, sorry. I've been having a difficult time keeping up with GitHub for the last couple months. School, and all that.

> Why? Because it's a fork?

No, because what they are trying to do is a bad idea.

> Why not?

Because people don't think in terms of protocols. Fundamentally, PRISM Break is about reaching people. So we have to present ideas in a form that make sense to people.

> If you ask people, I don't think they will say, that SMS is instant messaging.

I do. They're profoundly similar. They're both short, person-to-person exchanges that happen largely in real time. Virtually the only difference is that custom IM protocols often support "bonus features" and often can be synced across devices. Look at iMessage. It runs over the data connection, so I assume you'd consider it an instant messaging protocol, right? So if SMS isn't an IM protocol, how was Apple able to transparently upgrade SMS conversations to iMessage, where both recipients used iOS? It's because they're _the same medium._

> It there are no technical reasons not to recommend SMSSecure I see no reason not to recommend SMSSecure... If there was less discussion about personal likes and dislikes in this repo there won't be more than 200 open issues (each more than 10 comments). And there would be more people contributing interesting stuff...

I'm going to go ahead and assume that (while you didn't explicitly mention anyone) this comment was aimed at me.

I see where you're coming from. But you have to trust me when I say that this isn't personal for me. I have no personal vendetta against SMSSecure. I care a lot about the mass adoption of secure communication, and I care a lot about the potency of PRISM Break, now and in the future. And that reflects in the language that I use.

I've used stronger language against SMSSecure because I objectively think that it would be very bad to list it on PRISM Break, not because I e.g. hate the developers, or hate forking, or whatever.

On a technical level, I don't really have a problem with SMSSecure. I think it's pretty sound. But on the UX level, I have issues. And that's what I've been arguing about.

> BTW: Recently as TextSecure invented their new design, SMSSecure did this on the same day, so I think you can't say, they "failed on a user experience level"...

Just because you make something pretty doesn't mean that the underlying concepts and designs aren't fundamentally flawed.

User interface design and user experience design are very separate concepts. Sounds like SMSSecure improved their user interface design, which is awesome. I love seeing that in FLOSS. But that doesn't mean that their user _experience_ is fixed. And IMHO, their user experience cannot be fixed, because of all the reasons I've listed in previous comments and above.

**Comment by strugee** [comment]
> Maybe there are more tech and IT people motivated enough to read prism-break website than general public.

Maybe. But this is not the ideal end goal. We want PRISM Break to reach _everyone_.

**Comment by sedrubal** [comment]
I don't get it, why ux should be bad. It is working, it looks good it is easy to use and intuitive.

BTW: if ux is a criteria, why is tox listed? Tox doesn't really work, it's unintuitive and difficult to use (but looks good ;) ) and it's beta (by design). I like tox, too and I'm glad that this is listed, but if tox is listed there is no argument against SMSsecure.

**Comment by juanito003** [comment]
@sedrubal yes very good point! +1

**Comment by ximex** [comment]
I'm the same opinion like @sedrubal
The ux is better than some other projects.

**Comment by vyp** [comment]
Don't really agree that requiring key exchange should keep it off the list, if that's your main user experience argument. Especially when having SMS support makes it "special", not because *it would simply be an unique protocol on the list*, but because SMS does not require internet.

> We want PRISM Break to reach *everyone*.

So it would be useful to such people because they would otherwise send unencrypted (because they have no internet to use any of other listed applications).

**Comment by strugee** [comment]
> because they have no internet to use any of other listed applications

I can't think of any situation where someone would have SMS but no data.

> BTW: if ux is a criteria, why is tox listed?

Tox has the "experimental" flag. Surely you are not trying to argue that UX isn't an important part of PRISM Break?

See OpenWhisperSystems' [blog post on why they dropped SMS][1] support. tl;dr: key exchanges are ugly and **SMS leaks tons and tons of metadata**.

 [1]: https://whispersystems.org/blog/goodbye-encrypted-sms/

**Comment by strugee** [comment]
> Don't really agree that requiring key exchange should keep it off the list, if that's your main user experience argument.

Users shouldn't have to understand the concept of a key. Their apps should manage that for them.

**Comment by Dostoi** [comment]
"I can't think of any situation where someone would have SMS but no data."
Well I rarely activate my data connection to avoid draining my battery and I don't think I'm the only one but I can always receive SMS

**Comment by sedrubal** [comment]
>I can't think of any situation where someone would have SMS but no
>data.

Oh yes, there are situations. Think of rural areas and cheap mobile providers: there are many areas where mobile network doesn't work, even in Europe. And there are users without Google push on their Android... They are glad about old good SMS.

>Tox has the "experimental" flag.

I think, tox will still be experimental in 5 years, but that's another discussion.

>tl;dr

Thanks ;)

>key exchanges are ugly and **SMS leaks tons and tons of
>metadata**.

OK that's a strong argument. I don't know much about SMS beside the usability. Does it leak more metadata than mobile network? And does SMS only leak metadata on sending SMS? I know silent SMS and other mobile systems weaknesses, which you can't easily get rid of it, even if you don't use it.

>Users shouldn't have to understand the concept of a key

I think, users should get a basic understanding of that concept and it mustn't be completely intransparent (WhatsApp).

**Comment by sedrubal** [comment]
@Dostoy :+1:

**Comment by ximex** [comment]
@strugee i only have to travel in an other country. the costs for data connection is that damn high that i only would communicate via SMS (if no open WLAN is in range)

**Comment by vyp** [comment]
> I can't think of any situation where someone would have SMS but no data.

What do you mean? SMS uses GSM, so it can be used without data. And having no data is still very common, just like not having wifi everywhere you go. And there's the case as @ximex mentioned about those with no data plan, like me right now. It's almost like saying "I can't think of any situation where someone would have SMS but no internet connection."

Point is that even though you definitely don't want to be using SMS if you can otherwise, you also don't want to be sending unecrypted ones if you really need to.

**Comment by mastercoms** [comment]
But you can still message someone directly who has the app, right?

**Comment by vyp** [comment]
@mastercoms Provided they have a number I believe.

**Comment by theltalpha** [comment]
> key exchanges are ugly

So would you suggest to remove the GNU Privacy Guard (GnuPG) from Prism-break?

**Comment by strugee** [comment]
> So would you suggest to remove the GNU Privacy Guard (GnuPG) from Prism-break?

Eventually. Not in the near- or even medium-term future, because it's the best we can do right now and it has wide adoption. But yes, absolutely.

**Comment by strugee** [comment]
Also, IMHO, users are willing to put up with a lot more _stuff_ on a desktop. On their phones, users expect things to Just Work™. I don't have any data to back up that assertion, but that's my gut feeling.

**Comment by vyp** [comment]
> Also, IMHO, users are willing to put up with a lot more stuff on a desktop. On their phones, users expect things to Just Work™. I don't have any data to back up that assertion, but that's my gut feeling.

I understand but I don't think that's enough to dismiss the comparison to pgp. But what would you replace pgp with?

**Comment by strugee** [comment]
> @strugee i only have to travel in an other country. the costs for data connection is that damn high that i only would communicate via SMS (if no open WLAN is in range)

Not according to the OpenWhisperSystems blog post I linked above.

**Comment by strugee** [comment]
> But what would you replace pgp with?

Nothing, yet. That's why I said long-term. But PGP has a long, long list of problems. When it's ready, Pond looks nice.

**Comment by vyp** [comment]
> Not according to the OpenWhisperSystems blog post I linked above.

They're talking about third world countires. Doesn't mean it still doesn't apply.

**Comment by sedrubal** [comment]
Wow, stop! This was a rhethorical question, you can't answer "yes". - But that's an other thing.

> JustWorkTM

Ok, I understand that. But you have to try smssecure. It works as good as TextSecure (and Whatsapp, or however these apps are called, young people are used to use in this days ;) )

**Comment by vyp** [comment]
> Nothing, yet. That's why I said long-term. But PGP has a long, long list of problems. When it's ready, Pond looks nice.

So you want to keep SMSSecure off the list because in the long-term future there will be better options?

**Comment by vyp** [comment]
Thanks for mention of pond though. Link for readers (isn't exactly the easiest term to search for): https://pond.imperialviolet.org/

**Comment by strugee** [comment]
> Wow, stop! This was a rhethorical question, you can't answer "yes".

Why? You can't ask a rhetorical question and then get mad when someone calls you on it.

(I realize the above is phrased somewhat confrontationally, which isn't what I mean, but it's very late here and I can't figure out how to phrase it better.)

**Comment by strugee** [comment]
> young people are used to use in this days

I'll try SMSSecure. It's the least I can do. But I'll warn you that it's very difficult to pitch something where you have to even think about keys to young people.

And just so we're all on the same page, I'm 16, so when I talk about how to "pitch to young people", I know what I'm talking about.

**Comment by vyp** [comment]
> But I'll warn you that it's very difficult to pitch something where you have to even think about keys to young people.

Imo, it's not that it has to be perfect for everyone. I agree with you in that having to key exchange is clunky. It's simply culture and the vast majority of people will not bother/care/want to understand. But it's that it might be useful for a 'significant' minority of visitors. The downside is that visitors might see SMS as a safer/better option compared to sending messages via the internet (using better protocols that don't leak as much metadata), but my opinion is that the positives are still worth it.

**Comment by vyp** [comment]
@strugee I'm looking at pond (for the first time obviously) for like the past 5 minutes. And it looks real nice. But it seems to also require key exchanges before any communication can be done. That seems to be the *way* it's designed to work. In fact, it currently uses pgp for that. Am I missing something, because you can't really replace pgp with something that still uses pgp? You can though replace email *+* pgp with pond of course. But then your argument about requiring key exchanges with pgp (and hence removing it sometime in the future) doesn't seem to hold up?

**Comment by sedrubal** [comment]
**Off Topic**

**Comment by SecUpwN** [comment]
Have switched from `TextSecure` to `SMSSecure` since a few months now. **+1** for adding it! :+1:

**Comment by sedrubal** [comment]
#1268 Has been merged, so this is fixed...
